src/tooling/grain_cli.md

```diff
@@ -0,0 +1,96 @@
+| --maximum-memory-pages <size> | maximum number of WebAssembly memory pages                                  |
+| --compilation-mode <mode>     | compilation mode (advanced use only)                                        |
+| --elide-type-info             | don't include runtime type information used by toString/print               |
+| --release                     | compile using the release profile (production mode)                         |
+| --experimental-wasm-tail-call | enables tail-call optimization                                              |
+| --debug                       | compile with debugging information                                          |
+| --wat                         | additionally produce a WebAssembly Text (.wat) file                         |
+| --hide-locs                   | hide locations from intermediate trees. Only has an effect with `--verbose` |
+| --no-color                    | disable colored output                                                      |
+| --no-gc                       | turn off reference counting garbage collection                              |
+| --no-bulk-memory              | polyfill WebAssembly bulk memory instructions                               |
+| --wasi-polyfill <filename>    | path to custom WASI implementation                                          |
+| --use-start-section           | replaces the \_start export with a start section during linking             |
+| --no-link                     | disable static linking                                                      |
+| --no-pervasives               | don't automatically import the Grain Pervasives module                      |
+| --parser-debug-level <level>  | debugging level for parser output                                           |
+| --memory-base <addr>          | set the base address for the Grain heap                                     |
+| --source-map                  | generate source maps                                                        |
+| --strict-sequence             | enable strict sequencing                                                    |
+| --verbose                     | print critical information at various stages of compilation                 |
+| -h, --help                    | display help for command                                                    |
+
+### Example
+
+```sh
+grain compile hello.gr
+```
+
+## `grain run`
+
+Runs a WebAssembly file. The `grain run` command can run Grain programs compiled with `--no-link`, and can also run WebAssembly files produced by other compilers.
+
+### Example
+
+```sh
+grain run hello.gr.wasm
+```
+
+It should be noted that `grain run` is used here on the WebAssembly file produced via `grain compile`.
+
+## `grain doc`
+
+Produces documentation for a Grain program. See the full [Graindoc page](./graindoc) to learn more!
```

Comment:
Can you add the usage examples like done in `grain format`? Sometimes you just want a reference guide for running on CLI

Reply:
````suggestion
Produces documentation for a Grain program. See the full [Graindoc page](./graindoc) to learn more!

### Example

Generating docs for a single file to stdout:

```sh
grain doc hello.gr
```

Generating docs for a single file:

```sh
grain doc hello.gr -o hello.md
```

Generating docs for an entire project:

```sh
grain format . -o .
```
````

Something like this?

Reply:
LGTM!